lib/publiccloud/azure.pm

```diff
@@ -24,6 +24,7 @@ has container => 'sle-images';
 has image_gallery => 'test_image_gallery';
 has lease_id => undef;
 has storage_region => 'westeurope';
+has ssh_key => '/root/.ssh/id_rsa';
```

Comment:
why do we need to duplicate something already defined in provider ?

Reply:
if you need to redefine default for Azure I think it can be done in init before calling `$self->SUPER::init();` . I think this line is real example why people who saying that there is OOP in perl are wrong . This is something which "works" but does not make any sense from OOP perspective . You basically declaring **new** class field in a child which has same name as class field already defined in parent

Reply:
If i was going for the most "correct" solution, i would probably be overloading `new` i guess. 🤔 
```perl
sub new { shift->SUPER::new(ssh_key => '/root/.ssh/id_rsa') }
```

Reply:
ok , let's keep it like this . Just leave a comment above it that this is redefining of something defined in parent class